provider/consumer.py

```diff
@@ -96,32 +99,32 @@ def run(self):
-                logging.info("Firing trigger {} with {} messages with a total size of {} bytes".format(self.trigger, len(messages), messageSize))
-
-                try:
-                    response = requests.post(self.triggerURL, json=payload)
-                    status_code = response.status_code
-                    logging.info("Repsonse status code {}".format(status_code))
-
-                    if status_code >= 400 and status_code < 500:
-                        # TODO need to kill this trigger
-                        # However, 429 is the status code for a throttled response
-                        logging.info('Need to kill this trigger due to status code {}'.format(status_code))
-                    elif status_code >= 500:
-                        # TODO OW server error... now what?
-                        logging.error('Error talking to OW, status code {}'.format(status_code))
-                    else:
-                        # manually commit offsets only upon successfully
-                        # firing the trigger
-                        self.consumer.commit()
-                except requests.exceptions.RequestException as e:
-                    # TODO network issue talking to OW... now what?
-                    logging.error('Error talking to OW: {}'.format(e))
-
-                # TODO
-                # error handling
-                # retries? I suppose certain status codes warrant it...
-                # only commit offset if post was successful?
+
+                logging.info("Firing trigger {} with {} messages with a total size of {} bytes".format(self.trigger,
+                   len(messages), messageSize))
+
+                while retry:
+                    try:
+                        response = requests.post(self.triggerURL, json=payload)
+                        status_code = response.status_code
+                        logging.info("Repsonse status code {}".format(status_code))
+
+                        # Manually commit offset if the trigger was fired successfully. Retry firing the trigger if the
+                        # service was unavailable (503), an internal server error occurred (500), the request timed out
+                        # (408), or the request was throttled (429).
+                        if status_code == 200:
+                            self.consumer.commit()
+                            retry = False
+                        elif status_code != 503 or status_code != 500 or status_code != 408 or status_code != 429:
+                            logging.error('Error talking to OpenWhisk, status code {}'.format(status_code))
+                            retry = False
+
+                    except requests.exceptions.RequestException as e:
+                        logging.error('Error talking to OpenWhisk: {}'.format(e))
+
+                    if retry:
+                        logging.info("Failed to commit the offset. Retrying in {} second(s)".format(retry_timeout))
```

Comment:
This message isn't exactly correct because you didn't even attempt to commit the offset. Maybe just limit it to saying "Retrying in {} second(s)". Also please prefix all these new logging messages with the trigger name as `"[{}] Retrying in {} second(s)".format(self.trigger, retry_timeout)`